src/Microsoft.CodeQuality.Analyzers/UnitTests/ApiDesignGuidelines/CancellationTokenParametersMustComeLastTests.cs

```diff
@@ -238,6 +238,66 @@ public static void M1(this object p1, CancellationToken p2, object p3)
             VerifyCSharp(test, expected);
         }
 
+        [Fact, WorkItem(2281, "https://github.com/dotnet/roslyn-analyzers/issues/2281")]
+        public void CA1068_DoNotReportOnIProgressLastAndCancellationTokenBeforeLast()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+public class C
+{
+    public Task SomeAsync(object o, CancellationToken cancellationToken, IProgress<int> progress)
+    {
+        throw new NotImplementedException();
+    }
+}");
+
+            VerifyBasic(@"
+Imports System
+Imports System.Threading
+Imports System.Threading.Tasks
+
+Public Class C
+    Public Function SomeAsync(ByVal o As Object, ByVal cancellationToken As CancellationToken, ByVal progress As IProgress(Of Integer)) As Task
+        Throw New NotImplementedException()
+    End Function
+End Class");
+        }
+
+        [Fact, WorkItem(2281, "https://github.com/dotnet/roslyn-analyzers/issues/2281")]
+        public void CA1068_ReportOnIProgressLastAndCancellationTokenNotBeforeLast()
+        {
+            VerifyCSharp(@"
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+public class C
+{
+    public Task SomeAsync(CancellationToken cancellationToken, object o, IProgress<int> progress)
+    {
+        throw new NotImplementedException();
+    }
+}",
+            new DiagnosticResult(CancellationTokenParametersMustComeLastAnalyzer.Rule).WithLocation(8, 17)
+                .WithArguments("C.SomeAsync(System.Threading.CancellationToken, object, System.IProgress<int>)"));
+
+            VerifyBasic(@"
+Imports System
+Imports System.Threading
+Imports System.Threading.Tasks
+
+Public Class C
+    Public Function SomeAsync(ByVal cancellationToken As CancellationToken, ByVal o As Object, ByVal progress As IProgress(Of Integer)) As Task
+        Throw New NotImplementedException()
+    End Function
+End Class",
+            new DiagnosticResult(CancellationTokenParametersMustComeLastAnalyzer.Rule).WithLocation(7, 21)
+                .WithArguments("Public Function SomeAsync(cancellationToken As System.Threading.CancellationToken, o As Object, progress As System.IProgress(Of Integer)) As System.Threading.Tasks.Task"));
```

Comment:
The message is quite weird compared to C#, was it expected? I have seen that there isn't any test for VB.NET while the analyzer actually targets VB.NET.

Reply:
I'd say it's fine unless either:

1. We have specific guidelines for how to format signatures in messages like this, and this isn't following it
2. Someone is complaining about it and wants to propose a solution

Reply:
Either way, since it didn't _change_ as part of this pull request I would not expect it to need a fix here. 😄